habitat-lab/habitat/config/default_structured_configs.py

```diff
@@ -134,6 +137,14 @@ class StopActionConfig(ActionConfig):
     type: str = "StopAction"
 
 
+@attr.s(auto_attribs=True, slots=True)
+class VelocityStopActionConfig(ActionConfig):
```

Comment:
This is the third stop action defined here. Can this be RearrangeStopAction ? If not, why?

Reply:
This is the logic inside the step function of the rearrange_stop action.
```
    def step(self, task, *args, is_last_action, **kwargs):
        should_stop = kwargs.get("rearrange_stop", [1.0])
        if should_stop[0] > 0.0:
            rearrange_logger.debug(
                "Rearrange stop action requesting episode stop."
            )
            self.does_want_terminate = True

        if is_last_action:
            return self._sim.step(HabitatSimActions.rearrange_stop)
        else:
            return {}
```
This is the logic in the velocity_stop action:
```
        self, velocity_stop, task: EmbodiedTask, *args: Any, **kwargs: Any
    ):
        r"""Update ``_metric``, this method is called from ``Env`` on each
        ``step``.
        """
        if velocity_stop[0] > 0.0:
            task.is_stop_called = True  # type: ignore

        return self._sim.get_observations_at()  # type: ignore
```
The logic is mostly different between the two.
Also rearrange stop will have to be renamed if we want to use it in Habitat 1.0 tasks so it made sense to me to create a new action.

Reply:
@vincentpierre Does this answer your question